mlrun/config.py

```diff
@@ -937,6 +946,76 @@ def get_v3io_access_key(self):
         # Get v3io access key from the environment
         return os.environ.get("V3IO_ACCESS_KEY")
 
+    def get_model_monitoring_file_target_path(
+        self,
+        project: str = "",
+        kind: str = "",
+        target: str = "online",
+        artifact_path: str = None,
+    ) -> str:
+        """Get the full path from the configuration based on the provided project and kind.
+
+        :param project:        Project name.
+        :param kind:           Kind of target path (e.g. events, log_stream, endpoints, etc.)
+        :param target:         Can be either online or offline. If the target is online, then we try to get a specific
+                               path for the provided kind. If it doesn't exist, use the default path.
+                               If the target path is offline and the offline path is already a full path in the
+                               configuration, then the result will be that path as-is. If the offline path is a
+                               relative path, then the result will be based on the mlrun artifact path and the offline
+                               relative path.
+        :param artifact_path:  Optional artifact path that will be used as a relative path. If not provided, the
+                               relative artifact path will be taken from the global MLRun artifact path.
+
+        :return: Full configured path for the provided kind.
+        """
+
+        if target != "offline":
+            store_prefix_dict = (
+                mlrun.mlconf.model_endpoint_monitoring.store_prefixes.to_dict()
+            )
+            if store_prefix_dict.get(kind):
+                # Target exist in store prefix and has a valid string value
+                return store_prefix_dict[kind].format(project=project)
+            return mlrun.mlconf.model_endpoint_monitoring.store_prefixes.default.format(
+                project=project, kind=kind
+            )
+
+        # Get the current offline path from the configuration
+        file_path = mlrun.mlconf.model_endpoint_monitoring.offline_storage_path.format(
+            project=project, kind=kind
+        )
+
+        # Absolute path
+        if any(value in file_path for value in ["://", ":///"]) or os.path.isabs(
+            file_path
+        ):
+            if project not in file_path:
+                # Project name must be included
+                file_path = file_path.replace(kind, f"{project}/{kind}")
+            return file_path
+
+        # Relative path
+        else:
+            artifact_path = (
+                artifact_path
+                or mlrun.utils.helpers.fill_artifact_path_template(
+                    artifact_path=config.artifact_path, project=project
+                )
+            )
+            if artifact_path[-1] != "/":
+                artifact_path += "/"
+            if project not in file_path and project not in artifact_path:
```

Comment:
Again, I'm not too happy with this check. I'm also not sure this is needed if you already checked the same on the `file_path`